frontend/packages/knative-plugin/src/components/add/EventSourceForm.tsx

```diff
@@ -28,27 +29,31 @@ const EventSourceForm: React.FC<FormikProps<FormikValues> & OwnProps> = ({
   dirty,
   namespace,
   projects,
-}) => (
-  <Form className="co-deploy-image" onSubmit={handleSubmit}>
-    <EventSourcesSelector eventSourceList={useEventSourceList(namespace)} />
-    {values.type === EventSources.CronJobSource && <CronJobSection />}
-    {values.type === EventSources.SinkBinding && <SinkBindingSection />}
-    {values.type === EventSources.ApiServerSource && <ApiServerSection namespace={namespace} />}
-    <SinkSection namespace={namespace} />
-    <AppSection
-      project={values.project}
-      noProjectsAvailable={projects?.loaded && _.isEmpty(projects.data)}
-    />
-    <FormFooter
-      handleReset={handleReset}
-      errorMessage={status && status.submitError}
-      isSubmitting={isSubmitting}
-      submitLabel="Create"
-      sticky
-      disableSubmit={!dirty || !_.isEmpty(errors)}
-      resetLabel="Cancel"
-    />
-  </Form>
-);
+}) => {
+  return (
+    <Form className="co-deploy-image" onSubmit={handleSubmit}>
+      <EventSourcesSelector eventSourceList={useEventSourceList(namespace)} />
+      {values.type === EventSources.CronJobSource && <CronJobSection />}
+      {values.type === EventSources.SinkBinding && <SinkBindingSection />}
+      {values.type === EventSources.ApiServerSource && <ApiServerSection namespace={namespace} />}
+      {values.type === EventSources.CamelSource && <CamelSection />}
+      {values.editorType !== EditorType.YAML && <SinkSection namespace={namespace} />}
```

Comment:
You don't need to have this new property called `editorType` in values. You can simply inverse the above logic and render the contents only when the event source selected is in that white list.

Reply:
I kept the editor type property as to use in validation, as I needed to change the validation schema based on if it is a form or a yaml editor

Reply:
@nemesis09  validation Schema can be handled based on `type` right? 

something like 

```
const ownEvenstSources = Object.keys(EventSources);
export const eventSourceValidationSchema = yup.object().shape({
  project: yup.object().when('type', {
    is: (type) => ownEvenstSources.includes(type),
    then: projectNameValidationSchema,
  }),
......
})
```

Reply:
and just handle it based on type here as well as @rohitkrai03 commented remove `camelSource` from `EventSources` enum i.e if type is not in enum show `yamlEditor`